GoogleForCreators/web-stories-wp
# Editor: Add fundamentals for layer locking feature

## Context

This adds some fundamentals for layer locking feature. All changes are behind new feature flag.

## Non-user-facing changes

New option added to right-click menu:
| Element type | Right click menu |
|-|-|
| Media | ![Screen Shot 2022-04-26 at 19 50 21](https://user-images.githubusercontent.com/637548/165411183-a2fd6376-bb0b-4c41-99e1-f7258407a92f.png) |
| Shape | ![Screen Shot 2022-04-26 at 19 50 33](https://user-images.githubusercontent.com/637548/165411199-9c50769c-3518-4259-8927-f214b8b0a175.png) |
| Sticker | ![Screen Shot 2022-04-26 at 19 50 26](https://user-images.githubusercontent.com/637548/165411213-71041541-cd65-487d-bc27-6d505b1dbba0.png) |
| Text | ![Screen Shot 2022-04-26 at 19 50 40](https://user-images.githubusercontent.com/637548/165411238-e00b2b8a-9977-43be-97f0-354159f0c4dc.png) |

New functionality and design in layer panel (layer actions on hover):
| Locked? | Screenshot |
|-|-|
| 👍  | ![Screen Shot 2022-04-27 at 18 04 50](https://user-images.githubusercontent.com/637548/165639146-a98343fe-637c-4c78-b863-a6281c0d1e88.png) |
| 👎  | ![Screen Shot 2022-04-27 at 18 04 55](https://user-images.githubusercontent.com/637548/165639163-ea5fba45-2d95-40b1-8012-7f0e682e9f13.png) |

## Testing Instructions

This PR can be tested by following these steps:

1. Enable feature flag
2. Add elements of various type
3. Right click elements to toggle/untoggle lock
4. Right click layers to toggle/untoggle lock
5. Click the new lock icon on layer to toggle/untoggle lock
6. Observe that the layer correctly displays a lock icon if locked
7. Observe that no other changes has been made

## Checklist




---

Fixes #11368
